wdl/transforms/biscayne/src/main/scala/wdl/transforms/biscayne/linking/expression/values/BiscayneValueEvaluators.scala

```diff
@@ -0,0 +1,85 @@
+package wdl.transforms.biscayne.linking.expression.values
+
+import cats.data.NonEmptyList
+import cats.syntax.validated._
+import cats.syntax.traverse._
+import cats.instances.list._
+import common.validation.ErrorOr._
+import wdl.model.draft3.elements.ExpressionElement
+import wdl.model.draft3.elements.ExpressionElement.{AsMap, AsPairs, CollectByKey}
+import wdl.model.draft3.graph.expression.{EvaluatedValue, ForCommandInstantiationOptions, ValueEvaluator}
+import wdl.transforms.base.linking.expression.values.EngineFunctionEvaluators.processValidatedSingleValue
+import wom.expression.IoFunctionSet
+import wom.types._
+import wom.values.{WomArray, WomMap, WomPair, WomValue}
+import wom.types.coercion.defaults._
+
+object BiscayneValueEvaluators {
+  implicit val asMapFunctionEvaluator: ValueEvaluator[AsMap] = new ValueEvaluator[AsMap] {
+    override def evaluateValue(a: AsMap, inputs: Map[String, WomValue], ioFunctionSet: IoFunctionSet, forCommandInstantiationOptions: Option[ForCommandInstantiationOptions])
+                              (implicit expressionValueEvaluator: ValueEvaluator[ExpressionElement]): ErrorOr[EvaluatedValue[_ <: WomValue]] = {
+      processValidatedSingleValue[WomArray, WomMap](expressionValueEvaluator.evaluateValue(a.param, inputs, ioFunctionSet, forCommandInstantiationOptions)(expressionValueEvaluator)) {
+        case WomArray(WomArrayType(WomPairType(_: WomPrimitiveType, _)), values) =>
+          val validPairs: ErrorOr[List[(WomValue, WomValue)]] = values.toList traverse {
+            case WomPair(l, r) => (l, r).validNel
```

Comment:
[ToL] Does IntelliJ falsely error-highlight this for you, too? I couldn't figure out why it's upset.

Reply:
It's the `traverse` function. Apparently there's one-too-many steps of implicit induction for intellij to handle

Reply:
Sometimes adding a type ascription helps with these, but often it doesn't. 🤷‍♂️